tests/ut/ut_scargo_publish.py

```diff
@@ -0,0 +1,197 @@
+import os
+import subprocess
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+
+from scargo.commands.publish import (
+    conan_add_conancenter,
+    conan_add_remote,
+    conan_add_user,
+    conan_clean_remote,
+    scargo_publish,
+)
+from scargo.config import Config
+from tests.ut.utils import get_test_project_config
+
+
+@pytest.fixture
+def scargo_publish_test_setup(
+    monkeypatch: pytest.MonkeyPatch, tmp_path: Path
+) -> Config:
+    os.chdir(tmp_path)
+    monkeypatch.setattr("scargo.commands.publish.get_project_root", lambda: tmp_path)
+
+    test_project_config = get_test_project_config()
+    test_project_config.conan.repo["remote_repo_name_1"] = "https://some-url.io"
+    test_project_config.conan.repo["remote_repo_name_2"] = "https://some-url.io"
+
+    monkeypatch.setattr(
+        "scargo.commands.publish.prepare_config",
+        lambda: test_project_config,
+    )
+
+    return test_project_config
+
+
+def test_publish(
+    scargo_publish_test_setup: Config,
+    mock_subprocess_check_call: MagicMock,
+) -> None:
+    # ARRANGE
+    repo_name = "repo_name"
+
+    # ACT
+    scargo_publish(repo_name)
+
+    # ASSERT
+    project_config = scargo_publish_test_setup
```

Comment:
`project_config` is a misleading name for a `Config` instance, because there's also a `ProjectConfig` class.

Reply:
variable renamed